sql/sql_audit.cc

```diff
@@ -451,7 +451,48 @@ int mysql_audit_notify(THD *thd, mysql_event_general_subclass_t subclass,
   event.general_host= sctx->host();
   event.general_external_user= sctx->external_user();
   event.general_rows= thd->get_stmt_da()->current_row_for_condition();
-  event.general_sql_command= sql_statement_names[thd->lex->sql_command];
+  if(thd->lex->sql_command == SQLCOM_END && msg_len > 0 )
+  {
+    if(!strcmp(msg,"Prepare"))
```

Comment:
There is `command_name` array declared in `sql_parse.h`. You can iterate over its elements and find the index which you then will be able to cast to `COM_XXX` constant. After that you can `switch` over this constant and fill `event.general_sql_command` for known ones
```
  COM_STMT_PREPARE,
  COM_STMT_EXECUTE,
  COM_STMT_SEND_LONG_DATA,
  COM_STMT_CLOSE,
  COM_STMT_RESET,
  COM_STMT_FETCH,
  ...
```
with exactly the same `event.general_sql_command= sql_statement_names[<found_index>];`
This will help to remove `magic constants` from the code